core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -61,7 +61,16 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   var pyFiles: String = null
   var isR: Boolean = false
   var action: SparkSubmitAction = null
-  val sparkProperties: HashMap[String, String] = new HashMap[String, String]()
+  val sparkProperties: HashMap[String, String] = new HashMap[String, String]() {
+    override def put(k: String, v: String): Option[String] = {
```

Comment:
Bleh, can this be done without creating a subclass of `HashMap`? just a function that controls `put`-ting values in this map? CC @sryza on this one. I agree with the goal of making this consistent.